src/Model/mutators.ts

```diff
@@ -404,20 +406,23 @@ Model.prototype.add = function() {
   var segments = this._splitPath(subpath);
   return this._add(segments, value, cb);
 };
-Model.prototype.addPromised = promisify(Model.prototype.add);
+Model.prototype.addPromised = promisify<string>(Model.prototype.add);
 
 Model.prototype._add = function(segments, value, cb) {
   if (typeof value !== 'object') {
     var message = 'add requires an object value. Invalid value: ' + value;
     cb = this.wrapCallback(cb);
-    return cb(new Error(message));
+    cb(new Error(message));
+    return;
```

Comment:
return type 'undefined' is not assignable to type 'string'. consider adding  to your interface : string | undefined

Reply:
@ericyhwang what do you think of throwing an error here? Optional callback isn't great for the error handling case as this is really argument validation (so before any "work" that would result in callback call) but I'm unclear as to impacts or likelihood of this case. No id to return, and don't like the optional string return for downstream implications.

Reply:
FWIW Typescipt is okay with this (incompletely typed), and as this is what already exists in current code, It's what it will stay with for now.

Does highlight another point though... our type sigs allow `value: any` and not constrained to `object`